src/group_theory/perm/cycles.lean

```diff
@@ -4,12 +4,145 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Chris Hughes
 -/
 import group_theory.perm.sign
-import group_theory.order_of_element
 
 namespace equiv.perm
 open equiv function finset
+
 variables {α : Type*} {β : Type*} [decidable_eq α]
 
+section sign_cycle
+
+variables [fintype α]
+
+/-- A permutation is a cycle when any two nonfixed points of the permutation are related by repeated
+  application of the permutation. -/
+def is_cycle (f : perm β) := ∃ x, f x ≠ x ∧ ∀ y, f y ≠ y → ∃ i : ℤ, (f ^ i) x = y
+
+lemma is_cycle_swap {α : Type*} [decidable_eq α] {x y : α} (hxy : x ≠ y) : is_cycle (swap x y) :=
+⟨y, by rwa swap_apply_right,
+  λ a (ha : ite (a = x) y (ite (a = y) x a) ≠ a),
+    if hya : y = a then ⟨0, hya⟩
+    else ⟨1, by rw [gpow_one, swap_apply_def]; split_ifs at *; cc⟩⟩
+
+lemma is_cycle_inv {f : perm β} (hf : is_cycle f) : is_cycle (f⁻¹) :=
+let ⟨x, hx⟩ := hf in
+⟨x, by simp only [inv_eq_iff_eq, *, forall_prop_of_true, ne.def] at *; cc,
+  λ y hy, let ⟨i, hi⟩ := hx.2 y (by simp only [inv_eq_iff_eq, *, forall_prop_of_true, ne.def] at *; cc) in
+    ⟨-i, by rwa [gpow_neg, inv_gpow, inv_inv]⟩⟩
+
+lemma exists_gpow_eq_of_is_cycle {f : perm β} (hf : is_cycle f) {x y : β}
+  (hx : f x ≠ x) (hy : f y ≠ y) : ∃ i : ℤ, (f ^ i) x = y :=
+let ⟨g, hg⟩ := hf in
+let ⟨a, ha⟩ := hg.2 x hx in
+let ⟨b, hb⟩ := hg.2 y hy in
+⟨b - a, by rw [← ha, ← mul_apply, ← gpow_add, sub_add_cancel, hb]⟩
+
+lemma exists_pow_eq_of_is_cycle [decidable_eq β] [fintype β] {f : perm β}
```

Comment:
```suggestion
lemma exists_pow_eq_of_is_cycle {f : perm β}
```

Reply:
Those two hypotheses are required for `order_of`. The `fintype` hypotheses certainly cannot be removed (consider `n -> n+1` on the integers). I'm not familiar enough with `decidable_eq` to know if it's actually necessary.

Reply:
Ah, I see. You should still remove `decidable_eq` here to make the linter happy, likely by adding a `by classical; exact` or something to the proof